crates/noirc_evaluator/src/ssa/acir_gen/operations/intrinsics.rs

```diff
@@ -317,3 +318,11 @@ fn evaluate_println(
 
     evaluator.opcodes.push(AcirOpcode::Directive(log_directive));
 }
+
+fn format_field_string(field: FieldElement) -> String {
+    let mut trimmed_field = field.to_hex().trim_start_matches('0').to_owned();
+    if trimmed_field.len() % 2 != 0 {
```

Comment:
This is the first time I've heard of the string failing to parse if it has an odd number of digits. Seems like an odd limitation? Lets add a doc comment to this function explaining that

Reply:
From the `hex` crate:

```
/// The error type for decoding a hex string into `Vec<u8>` or `[u8; N]`.
#[derive(Debug, Clone, Copy, PartialEq)]
pub enum FromHexError {
    /// An invalid character was found. Valid ones are: `0...9`, `a...f`
    /// or `A...F`.
    InvalidHexCharacter { c: char, index: usize },

    /// A hex string's length needs to be even, as two digits correspond to
    /// one byte.
    OddLength,

    /// If the hex string is decoded into a fixed sized container, such as an
    /// array, the hex string's length * 2 has to match the container's
    /// length.
    InvalidStringLength,
}
```

Reply:
I think I will just copy their comment on the error

Reply:
This would be for parsing a hexadecimal string into an integer. This seems standard to what I've seen before, where you need two hexademical digits to represent a whole byte